File: Toy_Language_Interpreter/Repository/IRepo.java
package Repository;

import Exceptions.EmptyContainerException;
import Model.ProgramState;

import java.io.IOException;

public interface IRepo {
    ProgramState getCrtPrg() throws EmptyContainerException;
    boolean isEmpty();
    void add(ProgramState state);
    void logPrgStateExec(ProgramState state) throws IOException;
}

File: Toy_Language_Interpreter_C/Repository/IRepo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace a7
{
    interface IRepo
    {
        ProgramState GetCrtPrg();
        bool IsEmpty();
        void Add(ProgramState state);
        void LogPrgStateExec(ProgramState state);
    }
}
